Direct2D-Initialize

MMTsallagova opened the pull request:
- Add Direct2D Initialize
- Change device flag - BGRA_Support
- Add 2D lib
- Add test text

MMTsallagova committed:
Direct2D-Initialize

- Add Direct2D Initialize
- Change device flag - BGRA_Support
- Add 2D lib
- Add test text

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -94,6 +93,91 @@ bool Graphics::DirectXInitialize(int screenWidth, int screenHeight, HWND hWnd)
+bool Graphics::Direct2DInitialize(HWND hWnd)
+{
+    D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, &factory);
+    ID3D11Resource* res;
```
Какая-то странная логика разбиения строчек 
Объявление и получение ресурса должно быть вместе объединены (а они у тебя разделяются)
С surface тоже самое, у тебя объявление слито с ресурсом, а получение отдельно

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -94,6 +93,91 @@ bool Graphics::DirectXInitialize(int screenWidth, int screenHeight, HWND hWnd)
+    res->Release();
+    surface->Release();
+
+    HRESULT rs = renderTarget->CreateSolidColorBrush(D2D1_COLOR_F{ 0.4f,1.0f,1.0f,1.0f }, &rBrush);
```
Описание цвета - Пробел после запятой, а перед/после скобочки не нужен

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -94,6 +93,91 @@ bool Graphics::DirectXInitialize(int screenWidth, int screenHeight, HWND hWnd)
+    return false;
+}
+
+void Graphics::ConfigureBrush(FLOAT posX, FLOAT posY, const wchar_t* wszText_) // Configurate layout to text and brush to draw text
```
Комментарий строкой выше перенеси, у функций так вообще нельзя писать

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -2,7 +2,12 @@
 #include <d3d11_1.h>
+#include <d2d1.h>
+#include <dwrite.h>
 #include <iostream>
```
Раздели include и pragma

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -12,6 +17,11 @@ class Graphics
+    bool Direct2DInitialize(HWND hWnd);
+    bool DrawTextOnScene(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
+    void ConfigureBrush(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
+    void BeginDraw() { renderTarget->BeginDraw(); }
```
А что за методы BeginDraw и EndDraw? Зачем они? Они же не используются нигде и смысла в них нет

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -12,6 +17,11 @@ class Graphics
     bool DirectXInitialize(int screenWidth, int screenHeight, HWND hWnd);
+    bool Direct2DInitialize(HWND hWnd);
+    bool DrawTextOnScene(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
+    void ConfigureBrush(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
```
Нужно разделить public и private методы, а также отделить методы Direct2d от остальных методов

MMTsallagova replied:
Они все public, разве нет?

nikchernyakov replied:
Ты нигде не используешь ConfigureBrush в public
К слову, вызов инициализации Direct2d можно перенести внутрь инициализации всего DirectX, потому что это его компонент внутренний по факту

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -33,5 +43,13 @@ class Graphics
     ID3D11Texture2D* depthStencil = NULL;                     // Depth buffer texture
```
Null лучше заменить на nullptr

nikchernyakov replied:
И ниже также

MMTsallagova committed:
Direct2D-Initialize

- Direct2D initialization inside DirectX Initialization
- Fix grouping and paddings
- Move functions to private

nikchernyakov commented on Engine/Graphics.cpp:
```diff
@@ -91,9 +90,97 @@ bool Graphics::DirectXInitialize(int screenWidth, int screenHeight, HWND hWnd)
+    );
+
+}
+
```
Одна пустая строка разрыв между функциями должна быть

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -2,8 +2,14 @@
+#pragma comment(lib, "d2d1.lib") // This is what D2D1CreateFactory makes to work
+#pragma comment(lib, "Dwrite") // This is what DWriteCreateFactory makes to work
+
+
```
Одна строка пустая строка разрыв должен быть

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -23,15 +31,25 @@ class Graphics
 
-};
+    IDWriteFactory* pDWriteFactory_ = nullptr;
+    IDWriteTextFormat* pTextFormat_;
```
Только заметил, мы же договорились переменные без префикса "p" именовать
И убери нижнее подчёркивание в конце имён плиз
Эта строка и выше, а также у тебя в методе выше есть такое

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -23,15 +31,25 @@ class Graphics
 
+    bool Direct2DInitialize(HWND hWnd);
+    void ConfigureBrush(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
+};
```
В конце перед скобкой должен быть отступ

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -23,15 +31,25 @@ class Graphics
+    RECT rc;
 
+    bool Direct2DInitialize(HWND hWnd);
+    void ConfigureBrush(FLOAT posX, FLOAT posY, const wchar_t* wszText_);
```
Вот здесь последнюю переменную нужно пофиксить, убрать нижнее подчёркивание

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -23,15 +31,25 @@ class Graphics
-};
+    IDWriteFactory* pDWriteFactory_ = nullptr;
+    IDWriteTextFormat* pTextFormat_;
+    RECT rc;
```
Переименуй в rect, чтобы было очевиднее плиз

nikchernyakov commented on Engine/Graphics.h:
```diff
@@ -23,15 +31,25 @@ class Graphics
+    // Variables for Direct2D initialization
+    ID2D1Factory* factory = nullptr;
+    ID2D1RenderTarget* renderTarget;
+    ID2D1SolidColorBrush* rBrush = nullptr;
```
Убери r в начале плиз, чтобы brush было
Или в этом есть какая-то логика?

MMTsallagova committed:
Rename variables and fix code style
